website/content/docs/connect/ca/vault.mdx

```diff
@@ -30,45 +30,77 @@ must be met:
 
 ## Configuration
 
-The Vault CA is enabled by setting the `ca_provider` to `"vault"` and
-setting the required configuration values. An example configuration
-is shown below:
+The Vault CA is enabled by setting the CA provider to `"vault"` and
+setting the required configuration values.
+
+The configuration may either be provided in the agent's configuration file using
+the [`ca_provider`] and [`ca_config`]options, or configured using the
+[`/connect/ca/configuration`] API endpoint.
+
+Example configurations are shown below:
+
+<CodeTabs heading="Connect CA configuration" tabs={["Agent configuration", "API"]}>
+
+<CodeBlockConfig filename="/etc/consul.d/config.hcl" highlight="4,6-9">
 
 ```hcl
+# ...
 connect {
-    enabled = true
-    ca_provider = "vault"
-    ca_config {
-        address = "http://localhost:8200"
-        token = "..."
-        root_pki_path = "connect-root"
-        intermediate_pki_path = "connect-intermediate"
-    }
+  enabled = true
+  ca_provider = "vault"
+  ca_config {
+    address = "http://localhost:8200"
+    token = "..."
+    root_pki_path = "connect-root"
+    intermediate_pki_path = "connect-intermediate"
+  }
+}
+```
+
+</CodeBlockConfig>
+
+<CodeBlockConfig highlight="2,4-7">
+
+```json
+{
+  "Provider": "vault",
+  "Config": {
+    "Address": "http://localhost:8200",
+    "Token": "<token>",
+    "RootPKIPath": "connect-root",
+    "IntermediatePKIPath": "connect-intermediate"
+  }
 }
 ```
 
-The configuration options are listed below. Note, the
-first key is the value used in API calls and the second key (after the `/`)
-is used if you're adding configuring to the agent's configuration file.
+</CodeBlockConfig>
+
+</CodeTabs>
+
+The configuration options are listed below.
+
+-> **Note**: The first key is the value used in API calls, and the second key
+   (after the `/`) is used if you are adding the configuration to the agent's
+   configuration file.
 
 - `Address` / `address` (`string: <required>`) - The address of the Vault
   server.
 
 - `Token` / `token` (`string: <required>`) - A token for accessing Vault.
   This is write-only and will not be exposed when reading the CA configuration.
-  This token must have proper privileges for the PKI paths configured. In Consul
-  1.8.5 and later, if the token has the [renewable](https://www.vaultproject.io/api-docs/auth/token#renewable)
+  This token must have [proper privileges](#vault-acl-policies) for the PKI
+  paths configured. In Consul 1.8.5 and later, if the token has the [renewable](https://www.vaultproject.io/api-docs/auth/token#renewable)
   flag set, Consul will attempt to renew its lease periodically after half the
   duration has expired.
 
 - `RootPKIPath` / `root_pki_path` (`string: <required>`) - The path to
-  a PKI secrets engine for the root certificate. If the path doesn't
+  a PKI secrets engine for the root certificate. If the path does not
   exist, Consul will attempt to mount and configure this automatically.
```

Comment:
Not something we need to change in this PR, but I thought I would call out this issue about docs missing from this area: #10762

We don't say anything about what TTL Consul uses for the "configure this automatically" case.

Reply:
What are your thoughts on this updated description?

> The path to a PKI secrets engine for the root certificate. If the path does not exist, Consul will mount a new PKI secrets engine at the specified path with a [`max_lease_ttl`](https://www.vaultproject.io/api/system/mounts#max_lease_ttl) of 8760 hours, or 1 year (this TTL value is currently not configurable).

Reply:
Sounds good! It might also be good to say that the certificate expiry is set to 1 year as well. I'm not sure if most people will understand the implication that the `max_lease_ttl`  value is also the root CA expiry ttl.

Reply:
How about this phrasing?

> …of 8760 hours, or 1 year. This TTL value specifies the expiry period of the root certificate and is currently not configurable.

Reply:
Sounds great, thanks for making this change!